src/linear_algebra/matrix/diag.lean

```diff
@@ -0,0 +1,226 @@
+/-
```

Comment:
I think this file should be called `is_diag`, since we already have `matrix/diagonal.lean`